python/tank/platform/engine.py

```diff
@@ -1276,6 +1293,24 @@ def __debug_track_qt_widget(self, widget):
     ##########################################################################################
     # private and protected methods
 
+    def _emit_event(self, event):
+        """
+        Called by the engine whenever an event is to be emitted to child
+        apps of this engine.
+
+        .. note:: Events will be emitted and child apps notified immediately.
+
+        .. warning:: Some event types might be triggered quite frequently. Apps
+                     that react to events should do so in a way that is aware of
+                     the potential performance impact of their actions.
+
+        :param event: The event object that will be emitted.
+        :type event:  :class:`~sgtk.platform.events.EngineEvent`
+        """
+        self.log_debug("Emitting event: %s" % event)
+        for handler in self.__event_registry.get(event.__class__, []):
```

Comment:
Minor: Maybe worth adding a check that the event object passed in derives from EngineEvent. This will make it easier to support interface changes in the future without breaking clients' code I think. Makes things a little more strongly typed.